9.extension/README.md

```diff
@@ -0,0 +1,158 @@
+# Extending Dapr
+ 
+This sample explores the extendability of Dapr. There are use cases where you want to add custom functionality to Dapr, for example, you have an existing library logic like `Encryption/Decryption` that you would like to expose as an endpoint along with the Dapr API. In this sample you'll be creating a custom component `Crypto` and add it to the Daprd binary as a gRPC endpoint.
+ 
+In this sample you will be:
+ 
+- Creating a custom Daprd binary.
+- Use Dapr and Component-Contrib as libraries instead of binary.
+- Plugin custom gRPC endpoint to Dapr internal gRPC server.
+- Project structure and setup.
+ 
+Dapr can be extended by using itself as a library to create a new custom binary. For instance, create a new GO project, then copy original [main.go](https://github.com/dapr/dapr/blob/master/cmd/daprd/main.go) from Dapr Github source code to this new project(using Dapr as library here), next stitch together all the Components(Dapr runtime) in the same way as it was present in the original `main.go`, next register Custom component `Crypto` in runtime(we will explain this further). The new binary created after compilation is a custom `Daprd` binary with `Crypto` gRPC endpoint as a custom component.
+ 
+## Prerequisites
+ 
+For running this sample you should have the followings
+ 
+1. Knowledge about [Protocol-buffers](https://developers.google.com/protocol-buffers) 
```

Comment:
Not 100% possible but "Knowledge of" might be more appropriate?  If the other sample pages phrase it like this than keep it as is.